examples/AdvancedOperations/AddDisplayUploadAd.php

```diff
@@ -0,0 +1,243 @@
+use Google\Protobuf\StringValue;
+
+/**
+ * This code example adds a display upload ad to a given ad group.
+ * To get ad groups, run GetAdGroups.php.
+ */
+class AddDisplayUploadAd
+{
+    private const CUSTOMER_ID = 'INSERT_CUSTOMER_ID_HERE';
+    private const AD_GROUP_ID = 'INSERT_AD_GROUP_ID_HERE';
+
+    public static function main()
+    {
+        // Either pass the required parameters for this example on the command line, or insert them
+        // into the constants above.
+        $options = (new ArgumentParser())->parseCommandArguments(
+            [
+                ArgumentNames::CUSTOMER_ID => GetOpt::REQUIRED_ARGUMENT,
+                ArgumentNames::AD_GROUP_ID => GetOpt::REQUIRED_ARGUMENT
+            ]
+        );
+
+        // Generate a refreshable OAuth2 credential for authentication.
+        $oAuth2Credential = (new OAuth2TokenBuilder())->fromFile()->build();
+
+        // Construct a Google Ads client configured from a properties file and the
+        // OAuth2 credentials above.
+        $googleAdsClient = (new GoogleAdsClientBuilder())->fromFile()
+            ->withOAuth2Credential($oAuth2Credential)
+            ->build();
+
+        try {
+            self::runExample(
+                $googleAdsClient,
+                $options[ArgumentNames::CUSTOMER_ID] ?: self::CUSTOMER_ID,
+                $options[ArgumentNames::AD_GROUP_ID] ?: self::AD_GROUP_ID
+            );
+        } catch (GoogleAdsException $googleAdsException) {
+            printf(
+                "Request with ID '%s' has failed.%sGoogle Ads failure details:%s",
+                $googleAdsException->getRequestId(),
+                PHP_EOL,
+                PHP_EOL
+            );
+            foreach ($googleAdsException->getGoogleAdsFailure()->getErrors() as $error) {
+                /** @var GoogleAdsError $error */
+                printf(
+                    "\t%s: %s%s",
+                    $error->getErrorCode()->getErrorCode(),
+                    $error->getMessage(),
+                    PHP_EOL
+                );
+            }
+            exit(1);
+        } catch (ApiException $apiException) {
+            printf(
+                "ApiException was thrown with message '%s'.%s",
+                $apiException->getMessage(),
+                PHP_EOL
+            );
+            exit(1);
+        }
+    }
+
+    /**
+     * Runs the example.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the customer ID
+     * @param int $adGroupId the ad group ID to add a display upload ad to
+     */
+    public static function runExample(
+        GoogleAdsClient $googleAdsClient,
+        int $customerId,
+        int $adGroupId
+    ) {
+        // There are several types of display upload ads. For this example, we will create
+        // an HTML5 upload ad, which requires a media bundle.
+        // The DisplayUploadProductType field lists the available display upload types:
```

Comment:
I find this sentence misleading because it mixes `DisplayUploadProductType`, `DisplayUploadAdInfo` and `DisplayUploadAdInfo.display_upload_product_type`. What about only referencing `DisplayUploadProductType` here? More details about `DisplayUploadAdInfo` could then be provided in the  `createDisplayUploadAdGroupAd` function

Reply:
I think this's following golden CL :)

Reply:
I don't see why it's a problem here.
If a user follows the link, the user can see DisplayUploadProductType in there.
We can also educate users about both DisplayUploadProductType and DisplayUploadAdInfo at the same time, so I see this as advantage.

Reply:
SGTM